src/System.Buffers.Experimental/System/Buffers/BufferExtensions.cs

```diff
@@ -325,21 +325,11 @@ internal static int IndexOfStraddling(this ReadOnlySpan<byte> first, IReadOnlyBu
             // this check is a small optimization: if the first byte from the value does not exist in the bytesToSearchAgain, there is no reason to combine
             if (bytesToSearchAgain.IndexOf(value[0]) != -1)
             {
-                Span<byte> combined;
                 var combinedBufferLength = value.Length << 1;
-                if (combinedBufferLength < 128)
-                {
-                    unsafe
-                    {
-                        byte* temp = stackalloc byte[combinedBufferLength];
-                        combined = new Span<byte>(temp, combinedBufferLength);
-                    }
-                }
-                else
-                {
-                    // TODO (pri 3): I think this could be eliminated by chunking values
-                    combined = new byte[combinedBufferLength];
-                }
+                var combined = combinedBufferLength < 128 ?
```

Comment:
I think we should settle on the following idiom:
```c#
Span<byte> buffer = size < 128 ? stackalloc byte[size] : new byte[size];
```

.. as opposed to assigning to var and then having to cast.

Reply:
As it stands now the dominant type of a ternary expression is selected from one of consequence or alternative branches.  

If you have a span as a type of either of the branches, it will dominate. Otherwise compiler faces an ambiguity since no dominant type can be chosen.   

I will bring this up to language folks.  
There are some tradeoffs that we can have around the natural type of `stackalloc`.